File: code/2215.find-the-difference-of-two-arrays.php
<?php
/*
 * @lc app=leetcode id=2215 lang=php
 *
 * [2215] Find the Difference of Two Arrays
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums1
     * @param Integer[] $nums2
     * @return Integer[][]
     */
    function findDifference($nums1, $nums2) {
        return [
            array_unique(array_diff($nums1, $nums2)),
            array_unique(array_diff($nums2, $nums1)),
        ];
    }
}
// @lc code=end

File: code/2215.find-the-difference-of-two-arrays.go
package code

/*
 * @lc app=leetcode id=2215 lang=golang
 *
 * [2215] Find the Difference of Two Arrays
 */

// @lc code=start
func findDifference(nums1 []int, nums2 []int) [][]int {
	a, b := uniq(nums1), uniq(nums2)
	return [][]int{inOneNotInTwo(a, b), inOneNotInTwo(b, a)}
}

func uniq(x []int) map[int]bool {
	result := make(map[int]bool)
	for _, v := range x {
		result[v] = true
	}
	return result
}

func inOneNotInTwo(a, b map[int]bool) []int {
	result := make([]int, 0)
	for k := range a {
		if _, ok := b[k]; !ok {
			result = append(result, k)
		}
	}

	return result
}

// @lc code=end
